editor/src/clj/editor/collection_non_editable.clj

```diff
@@ -211,24 +209,27 @@
     (distinct)
     instance-property-descs))
 
-(defn- collection-desc->referenced-collection-proj-paths [collection-desc]
-  (eduction
-    (map :collection)
-    (distinct)
-    (:collection-instances collection-desc)))
+(defn- collection-desc->referenced-collection-proj-path->index [collection-desc]
+  (into {}
+        (comp (map :collection)
+              (distinct)
+              (map-indexed flipped-pair))
+        (:collection-instances collection-desc)))
 
-(defn- collection-desc->referenced-game-object-proj-paths [collection-desc]
-  (eduction
-    (map :prototype)
-    (distinct)
-    (:instances collection-desc)))
+(defn- collection-desc->referenced-game-object-proj-path->index [collection-desc]
+  (into {}
+        (comp (map :prototype)
+              (distinct)
+              (map-indexed flipped-pair))
+        (:instances collection-desc)))
 
-(defn- collection-desc->referenced-component-proj-paths [collection-desc]
-  (eduction
-    (map :data)
-    (mapcat game-object-non-editable/prototype-desc->referenced-component-proj-paths)
-    (distinct)
-    (:embedded-instances collection-desc)))
+(defn- collection-desc->referenced-component-proj-path->index [collection-desc]
+  (into {}
+        (comp (map :data)
+              (mapcat game-object-non-editable/prototype-desc->referenced-component-proj-paths)
+              (distinct)
+              (map-indexed flipped-pair))
+        (:embedded-instances collection-desc)))
```

Comment:
The same applies to any component resources (i.e. .sprite files or whatever) referenced from embedded game objects in our collection.